web/github.go

```diff
@@ -0,0 +1,109 @@
+package web
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/bradleyfalzon/ghinstallation"
+	"github.com/google/go-github/github"
+	"github.com/pressly/lg"
+)
+
+// GitHub is an HTTP service to call GitHub endpoints
+type GitHub struct {
+	AppID      int
+	PrivateKey string
+}
+
+func (g *GitHub) installations() ([]*github.Installation, error) {
+	appTr, err := ghinstallation.NewAppsTransportKeyFromFile(
+		http.DefaultTransport, g.AppID, g.PrivateKey)
+	if err != nil {
+		return nil, err
+	}
+
+	appClient := github.NewClient(&http.Client{Transport: appTr})
+
+	var installations []*github.Installation
+	opts := &github.ListOptions{
+		PerPage: 100,
+	}
+	for {
+		installs, resp, err := appClient.Apps.ListInstallations(context.TODO(), opts)
+		if err != nil {
+			return nil, err
+		}
+
+		installations = append(installations, installs...)
+
+		if resp.NextPage == 0 {
+			break
+		}
+
+		opts.Page = resp.NextPage
+	}
+
+	return installations, nil
+}
+
+// orgResponse is the response type used by Orgs handler
+type orgRespose struct {
+	Name string `json:"name"`
+}
+
+// Orgs writes in the response the list of organizations where the
+// logged-in user is an admin
+func (g *GitHub) Orgs(w http.ResponseWriter, r *http.Request) {
+	login, err := GetUserLogin(r.Context())
+	if err != nil {
+		lg.RequestLog(r).Errorf("failed to get user login from context: %s", err)
+		http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
+		return
+	}
+
+	installations, err := g.installations()
+	if err != nil {
+		lg.RequestLog(r).Errorf("failed to retrieve the GitHub App installations: %s", err)
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+
+	// initialized as empty array because otherwise json response will be null
+	// instead of []
+	orgs := []orgRespose{}
+
+	for _, installation := range installations {
+		// New transport for each installation
+		itr, err := ghinstallation.NewKeyFromFile(
+			http.DefaultTransport, g.AppID, int(installation.GetID()), g.PrivateKey)
+
+		if err != nil {
+			lg.RequestLog(r).Errorf("failed to initialize the GitHub App installation client: %s", err)
+			http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+			return
+		}
+
+		// Use installation transport in a new client
+		client := github.NewClient(&http.Client{Transport: itr})
+
+		if installation.Account == nil || installation.Account.Login == nil {
+			lg.RequestLog(r).Errorf("failed to get GitHub installation organization name")
+			http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+			return
+		}
+
+		org := *installation.Account.Login
```

Comment:
You need to check type here. For example logins of installations on our staging are:
- meysken
- carlosms
- campoy
- bzz
- bblfsh
- src-d

only last 2 are organizations

Reply:
Good catch, I only tested with an organization installation. Done in d067be8.

Now the question is... do we want to manage User installations in the web UI? I don't see why not, but at the same time I don't want to make the scope bigger right now. I'm thinking we can limit to organizations for now, and open an issue to check later, when all the functionality is done, if everything works for User installations. WDYT?

Reply:
Yes. Most probably there will be not many changes to support user installations too. It's okay to support it from the start or add later.

Reply:
I created an issue for this: #467.